modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/ServiceRegistry.java

```diff
@@ -42,23 +47,46 @@
    */
   public void inject(ServiceProvider serviceProvider) throws ServiceResolutionError {
     for (Field field : getAllFields(serviceProvider.getClass(), withAnnotation(Inject.class))) {
-      final Object dependency = lookup(field.getType());
-      if (dependency == null) {
-        throw new ServiceResolutionError(format("Cannot find a service to inject into field '%s' of service provider '%s'",
-                                                field.getName(), serviceProvider.getClass().getName()));
+      Class<?> dependencyType = field.getType();
+
+      boolean nullToOptional = false;
+      if (dependencyType.equals(Optional.class)) {
+        Type type = ((ParameterizedType) (field.getGenericType())).getActualTypeArguments()[0];
+        if (type instanceof ParameterizedType) {
```

Comment:
doesn't `GenericUtils` already have methods for doing this?

Reply:
No, I didn't find a helper method to avoid doing this here.